### DISCUSSION TITLE
2 AFC

### DISCUSSION THREAD

**Discussion by olgaslj** [jspsych/jsPsych#2424] [Q&A] [answered]
Hi,
I need to create a 2 alternative forced choice task, where a target and its foil appear simultaneously on the screen and participants decide which one they recognize from the training phase (that is always the target).
I have the input .js file where targets and foils are defined as different properties (e.g. {"target": "mouse", "foil": "touse"}).

This is how I wanted to implement this, but it's not working, it still shows only target.
var afc_test = {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: jsPsych.timelineVariable('target','foil'),
            choices: ['d','k']
        };

I'm new to javaScript, so I'm not sure if I'm getting the syntax wrong or the logic doesn't work.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @plavipaun,

You'll need to use a [dynamic parameter](https://www.jspsych.org/7.1/overview/dynamic-parameters/) to join together the two timeline variables. Here's a basic example that you could customize to fit your needs:

```js
var afc_test = {
  type: jsPsychHtmlKeyboardResponse,
  stimulus: function(){
    return `<p>${jsPsych.timelineVariable('target')}</p><p>${jsPsych.timelineVariable('foil')}</p>`;
  },
  choices: ['d','k']
};
```

You can adjust the appearance by changing the HTML/CSS of the string that is being returned.

**Reply by olgaslj** [reply]
Thanks a lot @jodeleeuw, it worked perfectly!
